os/binfmt/binfmt_loadbinary.c

```diff
@@ -169,12 +172,17 @@ int load_binary(int binary_idx, FAR const char *filename, load_attr_t *load_attr
 		if (bin->islibrary) {
 #ifdef CONFIG_ARMV7M_MPU
 #ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
-			mpu_get_register_value(NULL, MPU_REG_NUM_COM_LIB_TXT,  (uintptr_t)bin->alloc[ALLOC_TEXT], bin->textsize, true,  true);
-			mpu_get_register_value(NULL, MPU_REG_NUM_COM_LIB_RO,   (uintptr_t)bin->alloc[ALLOC_RO],   bin->rosize,   true,  false);
-			mpu_get_register_value(NULL, MPU_REG_NUM_COM_LIB_DATA, (uintptr_t)bin->alloc[ALLOC_DATA], bin->ramsize,  false, false);
+			/* Get MPU register values for MPU regions */
+			mpu_get_register_value(&com_bin_mpu_regs[3], MPU_REG_NUM_COM_LIB_TXT,  (uintptr_t)bin->alloc[ALLOC_TEXT], bin->textsize, true,  true);
+			mpu_get_register_value(&com_bin_mpu_regs[6], MPU_REG_NUM_COM_LIB_RO,   (uintptr_t)bin->alloc[ALLOC_RO],   bin->rosize,   true,  false);
+			mpu_get_register_value(&com_bin_mpu_regs[0], MPU_REG_NUM_COM_LIB_DATA, (uintptr_t)bin->alloc[ALLOC_DATA], bin->ramsize,  false, false);
```

Comment:
The order 3,6,0 is a mistake. 
Please change to 0,3,6.

Reply:
done